cla_backend/apps/cla_provider/tests/api/test_csvupload_api.py

```diff
@@ -389,6 +394,12 @@ def test_validator_valid_2018(self):
         validator = self.get_provider_csv_validator()
         self.assertEqual(len(validator.validate()), 2)
 
+    @override_settings(CONTRACT_2018_ENABLED=True)
+    def test_validator_is_2018_discrimination(self):
+        for matter_type in contract_2018_category_spec["discrimination"]["MATTER_TYPE1"]:
+            if get_applicable_contract(datetime.datetime(2019, 9, 1), matter_type) != CONTRACT_EIGHTEEN_DISCRIMINATION:
+                self.fail("Applicable contract is not 2018 discrimination contract")
```

Comment:
As an outsider, understanding the assertion here is a bit challenging. What does "applicable" mean?

Reply:
Different rates and fees apply to cases depending on what contract applies to them.
The different types of contracts are:

- CONTRACT_THIRTEEN
- CONTRACT_EIGHTEEN
- CONTRACT_EIGHTEEN_DISCRIMINATION

The value of Matter Type 1 field determines which contract applies to the current row.

Reply:
Hmm, it sounds like contracts apply to cases. Is that true? Or is the "provider" the contract holder? (In which case I would except a "provider" word appear somewhere in the test case.)

Reply:
I think in the wider business you are right that the provider is the contract holder. However the language used in some of tickets called the set of rules which a case is evaluated against a 'contract':
"As a Contract Manager I need to know whether a case uploaded within a Provider's csv upload has been completed under the 2018 Education contract " (LGA-608).

This PR and the highlighted fail message is about the case not validating against those rules.

Reply:
It feels there will be opportunities to simplify our models :) Not a blocker.

Reply:
(To clarify further) Would `2018 Education contract` mean a mixture of "provider has a 2018 contract", "education area", "case made while provider was under the 2018 contract"?

Reply:
Yes it has those meanings too. I think it can also mean:

A set of rules that apply to a case to qualify it as a 2018 Education contract case